test/e2e/main_test.go

```diff
@@ -59,6 +64,42 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
+// getEnvoyBinary reads E2E_GETENVOY_BINARY or defaults to "$PWD/build/bin/$GOOS/$GOARCH/getenvoy"
+// An error is returned if the value isn't an executable file.
+func getEnvoyBinary() (string, error) {
+	path := os.Getenv(E2E_GETENVOY_BINARY)
+	if path == "" {
+		// Assemble the default created by "make bin"
+		relativePath := filepath.Join("..", "..", "build", "bin", runtime.GOOS, runtime.GOARCH, "getenvoy")
+		abs, err := filepath.Abs(relativePath)
+		if err != nil {
+			return "", fmt.Errorf("resolve path to %s. Correct environment variable %s", path, E2E_GETENVOY_BINARY)
+		}
+		path = abs
+	}
+	stat, err := os.Stat(path)
+	if err != nil && os.IsNotExist(err) {
+		return "", fmt.Errorf("%s doesn't exist. Correct environment variable %s", path, E2E_GETENVOY_BINARY)
+	}
+	if stat.IsDir() {
+		return "", fmt.Errorf("%s is not a file. Correct environment variable %s", path, E2E_GETENVOY_BINARY)
+	}
+	// While "make bin" should result in correct permissions, double-check as some tools lose them, such as
+	// https://github.com/actions/upload-artifact#maintaining-file-permissions-and-case-sensitive-files
```

Comment:
I noticed some things in the shell scripts checking this, so figured I would move to enforce it here